### PR TITLE
Capture recent advice in our docs

### PR DESCRIPTION
**Pull request by theunrepentantgeek**
## What this PR does

Captures some advice recently given to a contributor for reference by others.

## How does this PR make you feel?

gif

## Checklist

- [x] this PR contains documentation

### CODE REVIEW

File: v2/tools/generator/internal/astmodel/type_name.go [hunk 1]
```diff
@@ -27,7 +27,6 @@ const (
 var armPackageDenyList = set.Make(
 	"appconfiguration",
 	"authorization",
-	"cache",
```
Comment (matthchr) [inline v2/tools/generator/internal/astmodel/type_name.go:30]: Did you mean for this to be in this PR? It seems unrelated to the title of the PR

File: docs/hugo/content/contributing/hints-and-tips/_index.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+### Combine as many resources as possible into a single call to CreateResources()
+
+One of the key features of ASO is that it takes care of sequencing - it works out the correct order of resource creation itself. This is essential, as when someone uses `kubectl` to apply a YAML file containing all the resources for their application, all the resurces get created in the cluster at the same times and ASO has to be able to do the right thing.
+
+It's important that we exercise this in our tests. We've found in the past some resources where additional work was required to make this run smoothly - this is why we have extension points defined in the [`genruntime/extensions package`](https://pkg.go.dev/github.com/Azure/azure-service-operator/v2@v2.10.0/pkg/genruntime/extensions).
+
+Instead of calling `tc.CreateResourceAndWait()` for each resource in turn, declare all the resources required for the test and then make a single call to `tc.CreateResourcesAndWait()` (note the plural in the name) to create them all at once. 
+
+
```
Comment (matthchr) [inline docs/hugo/content/contributing/hints-and-tips/_index.md:43]: minor: few excess newlines here

File: v2/tools/generator/internal/astmodel/type_name.go (hunk 1 above)
Comment (theunrepentantgeek) [reply v2/tools/generator/internal/astmodel/type_name.go]: Rebased without the extraneous changes.

File: docs/hugo/content/contributing/hints-and-tips/_index.md (hunk 2 above)
Comment (theunrepentantgeek) [reply docs/hugo/content/contributing/hints-and-tips/_index.md]: Removed.

**Review by matthchr — APPROVED** [review]
